packages/babel-plugin-import-jsx-pragma/src/index.js

```diff
@@ -0,0 +1,112 @@
+/**
+ * Default options for the plugin.
+ *
+ * @property {string}  scopeVariable Name of variable required to be in scope
+ *                                   for use by the JSX pragma. For the default
+ *                                   pragma of React.createElement, the React
+ *                                   variable must be within scope.
+ * @property {string}  source        The module from which the scope variable
+ *                                   is to be imported when missing.
+ * @property {boolean} isDefault     Whether the scopeVariable is the default
+ *                                   import of the source module.
+ */
+const DEFAULT_OPTIONS = {
+	scopeVariable: 'React',
+	source: 'react',
+	isDefault: true,
+};
+
+/**
+ * Babel transform plugin for automatically injecting an import to be used as
+ * the pragma for the React JSX Transform plugin.
+ *
+ * @see http://babeljs.io/docs/en/babel-plugin-transform-react-jsx
+ *
+ * @param {Object} babel Babel instance.
+ *
+ * @return {Object} Babel transform plugin.
+ */
+export default function( babel ) {
+	const { types: t } = babel;
+
+	let _options;
+
+	let hasJSX, hasImportedScopeVariable;
+
+	function getOptions( options ) {
+		if ( ! _options ) {
+			_options = {
+				...DEFAULT_OPTIONS,
+				...options,
+			};
+		}
+
+		return _options;
+	}
+
+	return {
+		visitor: {
+			JSXElement() {
+				hasJSX = true;
+			},
+			ImportDeclaration( path, state ) {
+				if ( hasImportedScopeVariable ) {
+					return;
+				}
+
+				const { scopeVariable, isDefault } = getOptions( state.opts );
+
+				// The module source isn't verified, since if at least the
+				// required variable is within scope, its assumed to be
+				// compatible with the targeted transform, and otherwise would
+				// conflict as a duplicate import if introduced separately.
+
```

Comment:
This newline is weird to me.

Reply:
>This newline is weird to me.

For context, there was previously a condition here which checked that the import source matched what we were expecting, but specifically I ran into issues with `@wordpress/element` where we import `createElement` from React proper:

https://github.com/WordPress/gutenberg/blob/04c0ea207efda81d98963c10e09c3c21705212fe/packages/element/src/index.js#L4-L15

I will update the comment and shift it down a line to describe the behavior as part of what it is we're actually testing now.